lib/dynamoid/indexes.rb

```diff
@@ -294,10 +294,10 @@ def validate_range_key
         if @range_key.present?
           range_field_attributes = @dynamoid_class.attributes[@range_key]
           if range_field_attributes.present?
-            range_key_type = range_field_attributes[:type]
-            if Dynamoid::Fields::PERMITTED_KEY_TYPES.include?(range_key_type)
+            range_key_type_dynamodb_type = dynamodb_type(range_field_attributes[:type], range_field_attributes)
+            if Dynamoid::Fields::PERMITTED_KEY_TYPES.include?(range_key_type_dynamodb_type)
```

Comment:
`PERMITTED_KEY_TYPES ` is a list of Dynamoid types, like date, datetime, and serialized. But `range_key_type_dynamodb_type` is a DynamoDB's type. So they are not completely compatible.

Reply:
Thanks for checking! Few options

1. `PERMITTED_KEY_TYPES` doesn't seem to be used anywhere else. We can declare it here and make it a list of dynamodb types instead of dynamoid types
2. Get the underlying type from the custom class similar to `PrimaryKeyTypeMapping.dynamodb_type`. Something like -
```
      def permitted_key_type?(dynamoid_key_type)
        if dynamoid_key_type.is_a?(Class)
          dynamoid_key_type = key_type.respond_to?(:dynamoid_field_type) ? type.dynamoid_field_type : :string
        end
        Dynamoid::Fields::PERMITTED_KEY_TYPES.include?(dynamoid_key_type)
      end

```

Let me know which one is preferrable and also if this feature that we would want to have.

Reply:
I would accept both options. But the first one seems a bit simpler. So I would prefer to have a list of allowed DynamoDB types.